src/net/tls.cc

```diff
@@ -529,6 +529,16 @@ void tls::certificate_credentials::set_dn_verification_callback(dn_callback cb)
     _impl->set_dn_verification_callback(std::move(cb));
 }
 
+template <typename T=gnutls_certificate_credentials_t>
```

Comment:
>Seconded. If you really want to access gnutls object, I would add an operator gnutls_certificate_credentials_t() const to credentials, and mark it as "may not exist, unsupported", because there is a reason we try to keep a slightly blackboxy thing here.

Makes sense. Do you think they'd value the templated/general interface you added over something explicit for gnutls (they are sorta saying exactly what they want?).

> Better still would be to expose abstracted accessors for all the aspects that seem relevant to the user.

How big a lift is this? We'd probably (1) only need to do it for exactly the information we need, (2) seems to be their preference and (3) kinda maintained the "slightly blackboxy" feel.

Reply:
I don't think it's a huge lift, but I think to do it without polluting the seastar header with gnutls types AND avoid the `void*` you'd need something similar to this anyway? Maybe I'm over-indexing on that aspect and `#include gnutls/gnutls.h` is what they're suggesting.

Reply:
I guess with direct accessors we're talking about not exposing gnutls types at all. Looking again at the usage I want in redpanda maybe that's totally fine.

Reply:
>I guess with direct accessors we're talking about not exposing gnutls types at all. Looking again at the usage I want in redpanda maybe that's totally fine.

yeh. i'd start with that: can we expose the information using standard types, and only when the answer becomes no, move towards exposing gnutls internals.

Reply:
ya, somewhat clunky, but reflective of the underlying gnutls APIs I guess. patch incoming